CMakeLists.txt

```diff
@@ -0,0 +1,469 @@
+    message(STATUS "compiling with tinydtls support")
+    set(HAVE_LIBTINYDTLS 1)
+  endif()
+
+  if(WITH_MBEDTLS)
+    set(WITH_MBEDTLS "${WITH_MBEDTLS}")
+    message(STATUS "compiling with mbedtls support")
+    # required libmbedtls (e.g. debian libmbedtls-dev)
+    set(HAVE_MBEDTLS 1)
+  endif()
+
+endif()
+
+message(STATUS "ENABLE_DTLS: ${ENABLE_DTLS}")
+message(STATUS "ENABLE_TCP: ${ENABLE_TCP}")
+message(STATUS "WITH_TINYDTLS: ${WITH_TINYDTLS}")
+message(STATUS "WITH_GNUTLS: ${WITH_GNUTLS}")
+message(STATUS "WITH_OPENSSL: ${WITH_OPENSSL}")
+message(STATUS "HAVE_LIBTINYDTLS: ${HAVE_LIBTINYDTLS}")
+message(STATUS "HAVE_LIBGNUTLS: ${HAVE_LIBGNUTLS}")
+message(STATUS "HAVE_OPENSSL: ${HAVE_OPENSSL}")
+message(STATUS "HAVE_MBEDTLS: ${HAVE_MBEDTLS}")
+
+if(${WITH_EPOLL}
+   AND ${HAVE_EPOLL_H}
+   AND ${HAVE_TIMERFD_H})
+  set(COAP_EPOLL_SUPPORT "1")
+  message(STATUS "compiling with epoll support")
+endif()
+
+if(ENABLE_SMALL_STACK)
+  set(ENABLE_SMALL_STACK "${ENABLE_SMALL_STACK}")
+  message(STATUS "compiling with small stack support")
+endif()
+
+set(PACKAGE_URL "https://libcoap.net/")
+set(PACKAGE_NAME "${PROJECT_NAME}")
+set(PACKAGE_TARNAME "${PROJECT_NAME}")
+set(PACKAGE_STRING "${PROJECT_NAME} ${PROJECT_VERSION}")
+set(PACKAGE_VERSION "${PROJECT_VERSION}")
+set(LIBCOAP_PACKAGE_VERSION "${PROJECT_VERSION}")
+set(LIBCOAP_PACKAGE_URL "${PACKAGE_URL}")
+set(LIBCOAP_PACKAGE_NAME "${PROJECT_NAME}")
+set(LIBCOAP_PACKAGE_STRING "TODO")
+set(LIBCOAP_PACKAGE_BUGREPORT "libcoap-developers@lists.sourceforge.net")
+set(PACKAGE_BUGREPORT "libcoap-developers@lists.sourceforge.net")
+set(top_srcdir "${CMAKE_CURRENT_LIST_DIR}")
+set(top_builddir "${CMAKE_CURRENT_BINARY_DIR}")
+set(COAP_DISABLE_TCP NOT "${ENABLE_TCP}")
+
+# creates config header file in build directory
+configure_file(${CMAKE_CURRENT_LIST_DIR}/include/coap2/coap.h.in ${CMAKE_CURRENT_BINARY_DIR}/include/coap2/coap.h)
+
+configure_file(${CMAKE_CURRENT_LIST_DIR}/include/coap2/coap_config.h.in
+               ${CMAKE_CURRENT_BINARY_DIR}/include/coap2/coap_config.h)
+
+#
+# library
+#
+
+set(COAP_LIBRARY_NAME "coap")
+
+add_library(
+  ${COAP_LIBRARY_NAME}
+  # sources
+  ${CMAKE_CURRENT_LIST_DIR}/src/pdu.c
+  ${CMAKE_CURRENT_LIST_DIR}/src/net.c
+  ${CMAKE_CURRENT_LIST_DIR}/src/coap_debug.c
+  ${CMAKE_CURRENT_LIST_DIR}/src/encode.c
+  ${CMAKE_CURRENT_LIST_DIR}/src/uri.c
+  ${CMAKE_CURRENT_LIST_DIR}/src/subscribe.c
+  ${CMAKE_CURRENT_LIST_DIR}/src/resource.c
+  ${CMAKE_CURRENT_LIST_DIR}/src/str.c
+  ${CMAKE_CURRENT_LIST_DIR}/src/option.c
+  ${CMAKE_CURRENT_LIST_DIR}/src/async.c
+  ${CMAKE_CURRENT_LIST_DIR}/src/block.c
+  ${CMAKE_CURRENT_LIST_DIR}/src/mem.c
+  ${CMAKE_CURRENT_LIST_DIR}/src/coap_io.c
+  ${CMAKE_CURRENT_LIST_DIR}/src/coap_session.c
+  ${CMAKE_CURRENT_LIST_DIR}/src/coap_hashkey.c
+  ${CMAKE_CURRENT_LIST_DIR}/src/address.c
+  ${CMAKE_CURRENT_LIST_DIR}/src/coap_time.c
+  ${CMAKE_CURRENT_LIST_DIR}/src/coap_event.c
+  ${CMAKE_CURRENT_LIST_DIR}/src/coap_notls.c
```

Comment:
Need to add in ${CMAKE_CURRENT_LIST_DIR}/src/coap_tcp.c

Reply:
Thanks, building works now.